src/interfaces/rtc_data_channel.cc

```diff
@@ -116,10 +116,16 @@ void RTCDataChannel::CleanupInternals() {
 }
 
 void RTCDataChannel::OnPeerConnectionClosed() {
+  // let channel clean up by itself. otherwise RTCPeerConnection has to react with the channel close event,
+  // to pop channel from __channels array, otherwise the freed channel pointer still be used in RTCPeerConnection,
+  // which will lead to crash.
+  // to avoid channel close event management in RTCPeerConnection, let's decouple the channel close and RTCPeerConnection.
+  /*
   if (_jingleDataChannel != nullptr) {
     CleanupInternals();
     Stop();
   }
+  */
```

Comment:
It might be acceptable to skip the call to CleanupInternals; however, Stop needs to be called, otherwise the event loop will be kept open and the process will not close. Can you please try building/running the tests?

Reply:
sure, will try.